packages/domain-search/src/components/suggestion-cta/index.tsx

```diff
@@ -41,12 +42,34 @@ export const DomainSuggestionCTA = ( { domainName }: DomainSuggestionCTAProps )
 				queries.domainAvailability( domainName )
 			);
 
+			if ( availability ) {
+				const isAvailable = DomainAvailabilityStatus.AVAILABLE === availability.status;
+				const isAvailableSupportedPremiumDomain =
+					DomainAvailabilityStatus.AVAILABLE_PREMIUM === availability.status &&
+					availability.is_supported_premium_domain;
+				// We only log the availability status if the domain is not available or not a supported premium domain
+				const unavailableStatus =
+					! isAvailable && ! isAvailableSupportedPremiumDomain ? availability.status : null;
+				events.onDomainAddAvailabilityPreCheck( unavailableStatus, domainName, suggestion.vendor );
+			}
+
 			if ( ! availability?.trademark_claims_notice_info ) {
 				return cart.onAddItem( suggestion );
 			}
 
 			setTrademarkClaimModalOpen( true );
 		},
+		onSuccess: ( data ) => {
+			// If `data` is undefined, the domain was not added to the cart yet
```

Comment:
Awesome. I would even go a step forward and return an object from the mutation with the `addedToTheCart` boolean. That would turn that implicit handling very explicit.

Reply:
This is still valid!